src/scanpy/preprocessing/_qc.py

```diff
@@ -298,9 +298,9 @@ def calculate_qc_metrics(
         )
     # Pass X so I only have to do it once
     X = _choose_mtx_rep(adata, use_raw=use_raw, layer=layer)
-    if isspmatrix_coo(X):
+    if isinstance(X, spmatrix) and not isinstance(X, _CSMatrix):
```

Comment:
This check seems different than the one before, and I'm not sure we want to be _more_ permissive

Reply:
Yeah, do we actually support `coo_matrix` values in AnnData? If not, we can probably just remove this code.

I made the change because if the code is “just coerce to `csr_matrix`”, there’s no reason to single out `coo_matrix`, we can just coerce everything that we know can be coerced.

Reply:
> Yeah, do we actually support coo_matrix values in AnnData? If not, we can probably just remove this code.

Very soon we will not actually:

```python
import numpy as np
import scipy.sparse as sp
import anndata as ad

arr = sp.random(10, 100, format="coo", random_state=np.random.default_rng())
ad.AnnData(X=arr)
```
yields
```
/Users/ilangold/Projects/Theis/anndata/src/anndata/_core/storage.py:48: FutureWarning: AnnData previously had undefined behavior around matrices of type <class 'scipy.sparse._coo.coo_matrix'>.In 0.12, passing in this type will throw an error. Please convert to a supported type.Continue using for this minor version at your own risk.
  warnings.warn(msg, FutureWarning)
```

Reply:
Then I’ll just go back to `coo`, letting anndata handle the warning.